packages/flutter/lib/src/cupertino/text_selection_toolbar.dart

```diff
@@ -946,66 +986,71 @@ class _RenderCupertinoTextSelectionToolbarItems extends RenderBox with Container
       return;
     }
 
+    // First pass: determine the height of the tallest child.
+    double greatestHeight = 0.0;
+    visitChildren((RenderObject renderObjectChild) {
+      final RenderBox child = renderObjectChild as RenderBox;
+      final double childHeight = child.getMaxIntrinsicHeight(constraints.maxWidth);
+      if (childHeight > greatestHeight) {
+        greatestHeight = childHeight;
+      }
+    });
+
     // Layout slotted children.
-    _backButton!.layout(constraints.loosen(), parentUsesSize: true);
-    _nextButton!.layout(constraints.loosen(), parentUsesSize: true);
+    final BoxConstraints slottedConstraints = BoxConstraints(
+      maxWidth: constraints.maxWidth,
+      minHeight: greatestHeight,
+      maxHeight: constraints.maxHeight,
```

Comment:
Sorry miss this part in the previous review. Should the height constraint here be updated too?